fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -339,9 +352,83 @@ public ParseNode visitSubmitTaskStatement(StarRocksParser.SubmitTaskStatementCon
         }
     }
 
-
     // ------------------------------------------- Materialized View Statement -----------------------------------------
 
+    @Override
+    public ParseNode visitCreateMaterializedViewStatement(
+            StarRocksParser.CreateMaterializedViewStatementContext context) {
+        if (!Config.enable_experimental_mv) {
+            throw new ParsingException("The experimental mv is disabled");
+        }
+        boolean ifNotExist = context.IF() != null;
+        QualifiedName qualifiedName = getQualifiedName(context.mvName);
+        TableName tableName = qualifiedNameToTableName(qualifiedName);
+        String comment =
+                context.comment() == null ? null : ((StringLiteral) visit(context.comment().string())).getStringValue();
+        // process query statement
+        QueryStatement queryStatement = (QueryStatement) visit(context.queryStatement());
+        // process properties
+        // get properties if exists
+        Map<String, String> properties = new HashMap<>();
+        if (context.properties() != null) {
+            List<Property> propertyList = visit(context.properties().property(), Property.class);
+            for (Property property : propertyList) {
+                properties.put(property.getKey(), property.getValue());
+            }
+        }
+        //process refresh
+        RefreshSchemeDesc refreshSchemeDesc = null;
+        if (context.refreshSchemeDesc() == null) {
+            refreshSchemeDesc = new SyncRefreshSchemeDesc();
+        } else {
+            refreshSchemeDesc = ((RefreshSchemeDesc) visit(context.refreshSchemeDesc()));
+        }
+        if (refreshSchemeDesc instanceof SyncRefreshSchemeDesc) {
+            if (context.primaryExpression() != null) {
+                throw new IllegalArgumentException(
+                        "Partition by is not supported by SYNC refresh type int materialized view");
+            }
+            if (context.distributionDesc() != null) {
+                throw new IllegalArgumentException(
+                        "Distribution by is not supported by SYNC refresh type in materialized view");
+            }
+            String sql = AST2SQL.toString(queryStatement);
+            StatementBase statement = SqlParser.parseWithOldParser(sql, sqlMode, 0);
```

Comment:
why use old parser?

Reply:
@HangyuanLiu suggest to compatible old grammar  in first phase,  we were replace this logic after sync logic added .